matching_test.go

```diff
@@ -0,0 +1,29 @@
+package backoff
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+var (
+	customError    = fmt.Errorf("custom error")
+	differentError = fmt.Errorf("different error")
+
+	tests = map[error]error{
+		// sanity check
+		customError:    customError,
+		differentError: differentError,
+		// actual test
+		Permanent(customError):    customError,
+		Permanent(differentError): differentError,
+	}
+)
+
+func Test_ErrorMatching(t *testing.T) {
+	for err, expectedType := range tests {
+		if !errors.Is(err, expectedType) {
```

Comment:
For my understanding, with this we still need to use `microerror.Cause` in our error matchers. So what does really get fixed here? Is it that we deprecated `microerror.Cause` in favour of the new native `errors.Is`? Because then we have to rewrite everything I guess.

Reply:
This is a topic for a larger discussion, I guess. `microerror.Cause` is marked as deprecated since https://github.com/giantswarm/microerror/pull/22 with recommendation to use `errors` package. No clear migration path was laid out when it was merged and I do not feel I'm the right person to chart it out.

The goal of this PR is "let's make sure nobody has to find out that `backoff.Permanent` does not work with `errors` package the hard way". I've found this issue during a fairly innocent change. I can imagine the amount of errors it would produce during said migration.

I guess whenever we release next `backoff` release would be a good time to talk about releasing `microerror` without `Cause`. Then the migration can happen organically.